modules/giapi/src/main/scala/giapi/client/GiapiDb.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package giapi.client
+
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import cats.Applicative
+import monocle.Lens
+import monocle.function.At.at
+import monocle.function.At.atMap
+
+sealed trait StatusValue extends Product with Serializable
+
+object StatusValue {
+  final case class IntValue(value:    Int) extends StatusValue
+  final case class StringValue(value: String) extends StatusValue
+  final case class FloatValue(value:  Float) extends StatusValue
+  final case class DoubleValue(value: Double) extends StatusValue
+}
+
+/////////////////////////////////////////////////////////////////
+// The GiapiDb stores values streamed from the GMP
+/////////////////////////////////////////////////////////////////
+trait GiapiDb[F[_]] {
+
+  def value(i: String): F[Option[StatusValue]]
+
+  def update[A: ItemGetter](i: String, s: A): F[Unit]
+}
+
+object GiapiDb {
+  def newDb[F[_]: Sync](): F[GiapiDb[F]] =
```

Comment:
I must say that the return type here completely confuses me. 😕

Reply:
You don't need an empty argument list here.

Reply:
@sraaphorst the return type says, give me a program in `F` that yields a `GiapiDb[F]`; i.e., a `GiapiDb` whose operations are also defined in the effect `F`. You end up with a type like this when you need to do some work in `F` in order to construct the yielded value. This might not make sense, sorry. I did a talk at Scala Days on effects that you might find helpful if the terminology is confusing. Or ask more questions obvs.